include/pros/adi.hpp

```diff
@@ -18,42 +18,66 @@
 #ifndef _PROS_ADI_HPP_
 #define _PROS_ADI_HPP_
 
-#include "pros/adi.h"
-
 #include <cstdint>
+#include <tuple>
+#include <utility>
+
+#include "pros/adi.h"
 
 namespace pros {
+
+/** type definition for the pair of smart port and adi port for the basic adi devices */
+using ext_adi_port_pair_t = std::pair<std::uint8_t, std::uint8_t>;
+
+/** type definition for the triplet of smart port and two adi ports for the two wire adi devices*/
+using ext_adi_port_tuple_t = std::tuple<std::uint8_t, std::uint8_t, std::uint8_t>;
+
 class ADIPort {
 	public:
 	/**
 	 * Configures an ADI port to act as a given sensor type.
 	 *
 	 * This function uses the following values of errno when an error state is
 	 * reached:
-	 * ENXIO - The given value is not within the range of ADI Ports
+	 * ENXIO - The given value is not within the range of ADI Ports, or the Smart Port is not within the valid range
```

Comment:
3 things:

- 80-column limit is enforced for comments
- don't capitalize smart port like that
- "the given value is not within the range of ADI ports" both ADI port and smart port values are given as values

```suggestion
	 * ENXIO - Either the ADI port value or the smart port value is not within its
 		   valid range (ADI port: 1-8, 'a'-'h', or 'A'-'H'; smart port: 1-21).
```

Reply:
needless to say, this should apply for all subsequent occurrences of this line